Mathlib/FieldTheory/Tower.lean

```diff
@@ -116,11 +117,6 @@ theorem right [hf : FiniteDimensional F A] : FiniteDimensional K A :=
     exact Submodule.subset_span⟩⟩
 #align finite_dimensional.right FiniteDimensional.right
 
-/-- Tower law: if `A` is a `K`-vector space and `K` is a field extension of `F` then
-`dim_F(A) = dim_F(K) * dim_K(A)`. -/
-theorem finrank_mul_finrank : finrank F K * finrank K A = finrank F A := finrank_mul_finrank' F K A
```

Comment:
`finrank_mul_finrank'` is strictly more general as you can see from this proof. I removed `finrank_mul_finrank` and gave its name to the old `finrank_mul_finrank'`.

Reply:
Unfortuantely, removing `finrank_mul_finrank` led to typeclass synthesization failure, so I forced pushed to revert the commit.

Reply:
I think we should investigate what happened at the synthesization failure rather than keeping duplicated lemmas.

Reply:
Or at least replace this with a scarier name and point to the primed version in the docstring.
(maybe that should be unprimed and this be called something like `finrank_mul_finrank_of_field`)

Reply:
The failure was [here](https://github.com/leanprover-community/mathlib4/actions/runs/7437424982/job/20235046209):
```
[3992/4084] Buildierror: > LEAN_PATH=./.lake/packages/std/.lake/build/lib:./.lake/packages/Qq/.lake/build/lib:./.lake/packages/aesop/.lake/build/lib:./.lake/packages/proofwidgets/.lake/build/lib:./.lake/packages/Cli/.lake/build/lib:./.lake/build/lib LD_LIBRARY_PATH=./.lake/build/lib /home/lean/.elan/toolchains/leanprover--lean4---v4.5.0-rc1/bin/lean -Dpp.unicode.fun=true -Dpp.proofs.withType=false -DautoImplicit=false -DrelaxedAutoImplicit=false ./././Mathlib/FieldTheory/PolynomialGaloisGroup.lean -R ././. -o ./.lake/build/lib/Mathlib/FieldTheory/PolynomialGaloisGroup.olean -i ./.lake/build/lib/Mathlib/FieldTheory/PolynomialGaloisGroup.ilean -c ./.lake/build/ir/Mathlib/FieldTheory/PolynomialGaloisGroup.c
error: stdout:
Error: ./././Mathlib/FieldTheory/PolynomialGaloisGroup.lean:425:10: error: failed to synthesize
  AddCommGroup ↥F⟮α⟯
(deterministic) timeout at 'typeclass', maximum number of heartbeats (20000) has been reached (use 'set_option synthInstance.maxHeartbeats <num>' to set the limit)
```
the line is `rw [← FiniteDimensional.finrank_mul_finrank F F⟮α⟯ p.SplittingField,`
I guess it's the well-known problem that Lean fails to synthesize `AddCommGroup F⟮α⟯` from stronger classes for subojbects like `F⟮α⟯`. It looks for `AddCommGroup` because there's a `Module.Free` condition in `finrank_mul_finrank'` which is not present (automatic) in `finrank_mul_finrank`. I'll add a docstring to explain this.

Reply:
> (maybe that should be unprimed and this be called something like finrank_mul_finrank_of_field)

I think I'd rather rename `finrank_mul_finrank'` to `finrank_mul_finrank_of_free`. Would you like me to do that?

Reply:
couldn't a `letI` or a `maxHeartbeat` fix that?
I still think the ring version should be the main lemma as it is strictly more general and mirrors `rank_mul_rank`.

Reply:
Okay, I'm [trying this again](https://github.com/leanprover-community/mathlib4/pull/9046/commits/f87e8a4383095ab8f9fab65adb9cccac6979b3de) by fixing the timeout with `letI _ : AddCommGroup F⟮α⟯ := Ring.toAddCommGroup`. Hopefully no other errors pop up. But why not keep a lemma that is easier to use (can be used with less manual instances)?

Reply:
First of all it shouldn't have needed manual instances. It is a bug and hopefully will be fixed.
Second, I'm not totally against such duplication, but like `FiniteDimensional.finrank_eq_rank'`, but I think the duplication should be primed and the docstring should clearly say that it is a duplication for performance reasons.